src/lib/route-broadcaster.js

```diff
@@ -0,0 +1,126 @@
+'use strict'
+
+const co = require('co')
+const defer = require('co-defer')
+const _ = require('lodash')
+const request = require('co-request')
+const BROADCAST_INTERVAL = 30 * 1000 // milliseconds
+
+class RouteBroadcaster {
+  /**
+   * @param {RoutingTables} routingTables
+   * @param {Backend} backend
+   * @param {Object} config
+   * @param {Object} config.ledgerCredentials
+   * @param {Object} config.tradingPairs
+   * @param {Number} config.minMessageWindow
+   */
+  constructor (routingTables, backend, config) {
+    this.baseURI = routingTables.baseURI
+    this.routingTables = routingTables
+    this.backend = backend
+    this.ledgerCredentials = config.ledgerCredentials
+    this.tradingPairs = config.tradingPairs
+    this.minMessageWindow = config.minMessageWindow
+    this.adjacentConnectors = {}
+    this.adjacentLedgers = {}
+    for (const pair of config.tradingPairs) {
+      const destinationLedger = pair[1].split('@')[1]
+      this.adjacentLedgers[destinationLedger] = true
+    }
+  }
+
+  * start () {
+    yield this.crawlLedgers()
+    yield this.reloadLocalRoutes()
+    yield this.broadcast()
+    setInterval(() => this.routingTables.removeExpiredRoutes(), 1000)
+    defer.setInterval(() => {
+      return this.reloadLocalRoutes().then(this.broadcast.bind(this))
+    }, BROADCAST_INTERVAL)
+  }
+
+  broadcast () {
+    const routes = this.routingTables.toJSON()
+    return Promise.all(
+      Object.keys(this.adjacentConnectors).map(
+        (adjacentConnector) => this._broadcastTo(adjacentConnector, routes)))
+  }
+
+  _broadcastTo (adjacentConnector, routes) {
+    return request({
+      method: 'POST',
+      uri: adjacentConnector + '/routes',
+      body: routes,
+      json: true
+    }).then((res) => {
+      if (res.statusCode !== 200) {
+        throw new Error('Unexpected status code: ' + res.statusCode)
+      }
+    })
+  }
+
+  crawlLedgers () {
+    return Object.keys(this.adjacentLedgers).map(this._crawlLedger, this)
+  }
+
+  * _crawlLedger (ledger) {
+    const res = yield request({
```

Comment:
This should use the multiledger abstraction (if you don't want to change it here we can also make that change when everything is switched over to using the ledger plugin framework)